server/src/api/modules/competitions/competition.service.js

```diff
@@ -364,10 +368,13 @@ async function edit(id, title, metric, startsAt, endsAt, participants, verificat
 
   if (participants) {
     // Check if every username in the list is valid
-    for (let i = 0; i < participants.length; i += 1) {
-      if (!playerService.isValidUsername(participants[i])) {
-        throw new BadRequestError(`Invalid player username: ${participants[i]}`);
-      }
+    const usernamesCheck = await playerService.isValidUsernames(participants);
```

Comment:
Maybe this variable could be renamed to "invalidUsernames"?

**Also: isValidUsernames is not an async function so it does not need the "await" before calling it.**

Reply:
I'll rename it.